examples/update-on-async-rendering/initializing-state-before.js

```diff
@@ -0,0 +1,12 @@
+// Before
+class ExampleComponent extends React.Component {
+  state = {};
+
+  // highlight-range{1-6}
+  componentWillMount() {
+    this.setState({
+      isScrollingDown: false,
```

Comment:
I was using the relationship between `props.isScrollingDown` and `props.currentRow` in the other example specifically to explain what “derived state” means. 

I think using the same names here makes that example less useful. This particular example also doesn’t quite make sense (why would the “last” row be equal to the current row in props?)

I propose each example be different and show a plausible pattern. For example this was one could just be

```js
this.setState({
  currentColor: this.props.defaultColor,
  palette: 'rgb'
});
```